main/src/cgeo/geocaching/connector/gc/GCParser.java

```diff
@@ -1952,4 +1952,27 @@ public static String getUsername(final String page) {
         return StringUtils.isNotEmpty(usernameOld) ? usernameOld : null;
     }
 
+    @Nullable
+    public static int getCachesCount(final String page) {
+        int cachesCount = -1;
+        try {
+            String intStringToParse = removeDotAndComma(TextUtils.getMatch(page, GCConstants.PATTERN_CACHES_FOUND_LOGIN_PAGE, true, ""));
```

Comment:
Is there a chance to have a parameter to this method with the type of page (login or other) to directly call the right parser?

Reply:
I will check that

Reply:
seems that the login-page also has the headerSettings - I will have a closer look at this

Reply:
Thanks for the hint. Sometimes you stick to one approach and don't see another one.

Reply:
for the username I switched to `headerSettings` completely, as meanwhile (or maybe only on subpages) the `dataLayer` got extracted into calling an external script

Reply:
ok, then switching copmpletly to headerSettings for findCount would be safe as well